src/scenic/core/object_types.py

```diff
@@ -1255,67 +1256,138 @@ def boundingBox(self):
+            ).boundingPolygon.inradius
+
+        # Define a helper function that computes the support of the inradius,
+        # given the sub supports.
+        def planarInradiusSupport(width_s, length_s, shape_s):
+            # Unpack the dimension supports (and ignore the shape support)
+            min_width, max_width = width_s
+            min_length, max_length = length_s
+
+            if None in [min_width, max_width, min_length, max_length]:
+                # Can't get a bound on one or more dimensions, abort
+                return None, None
+
+            min_bounds = np.array([min_width, min_length, 1])
+            max_bounds = np.array([max_width, max_length, 1])
+
+            # Extract a list of possible shapes
+            if isinstance(self.shape, Shape):
+                shapes = [self.shape]
+            elif isinstance(self.shape, MultiplexerDistribution) and all(
+                isinstance(opt, Shape) for opt in self.shape.options
+            ):
+                shapes = self.shape.options
             else:
                 # Something we don't recognize, abort
-                return 0
-
-        # Check that all possible shapes contain the origin
-        if not all(shape.containsCenter for shape in shapes):
-            # One or more shapes has inradius 0
-            return 0
+                return None, None
+
+            # Get the inradius of the projected for each shape with the min and max bounds
+            min_distances = [
+                MeshVolumeRegion(
+                    mesh=shape.mesh, dimensions=min_bounds
+                ).boundingPolygon.inradius
+                for shape in shapes
+            ]
+            max_distances = [
+                MeshVolumeRegion(
+                    mesh=shape.mesh, dimensions=max_bounds
+                ).boundingPolygon.inradius
+                for shape in shapes
+            ]
 
-        # Get the inradius for each shape with the min and max bounds
-        min_distances = [
-            MeshVolumeRegion(mesh=shape.mesh, dimensions=min_bounds).inradius
-            for shape in shapes
-        ]
-        max_distances = [
-            MeshVolumeRegion(mesh=shape.mesh, dimensions=max_bounds).inradius
-            for shape in shapes
-        ]
+            distance_range = (min(min_distances), max(max_distances))
 
-        distance_range = (min(min_distances), max(max_distances))
+            return distance_range
 
-        return InradiusHelper(support=distance_range)
+        # Return either the inradius or a FunctionDistribution using the above helpers
+        args = toDistribution((self.width, self.length, self.shape))
+        if not isLazy(args):
+            return planarInradiusActual(*args)
+        else:
+            return FunctionDistribution(
+                args=args,
+                kwargs={},
+                func=planarInradiusActual,
+                support=planarInradiusSupport,
+            )
```

Comment:
Same comment as above. Also maybe not for this PR, but it would be nice to consolidate the logic here, since it's quite similar between these two values. Maybe we can make a decorator or helper function for properties which are monotonic in the object dimensions.